server/src/test/java/io/druid/server/initialization/JettyTest.java

```diff
@@ -283,4 +287,21 @@ public void testExtensionAuthFilter() throws Exception
     get.setRequestProperty(DummyAuthFilter.AUTH_HDR, "hacker");
     Assert.assertEquals(HttpServletResponse.SC_UNAUTHORIZED, get.getResponseCode());
   }
+
+  @Test
+  public void testRequestDecompression() throws Exception
+  {
+    String text = "hello";
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    try (GZIPOutputStream gzipOutputStream = new GZIPOutputStream(out)) {
+      gzipOutputStream.write(text.getBytes(Charset.defaultCharset()));
+    }
+    Request request = new Request(HttpMethod.POST, new URL("http://localhost:" + port + "/decompression"));
+    request.setHeader("Content-Encoding", "gzip");
+    request.setContent(MediaType.TEXT_PLAIN, out.toByteArray());
+    Assert.assertEquals(text, new String(IOUtils.toByteArray(client.go(
```

Comment:
Can you add a test where the request has `Accept-Encoding:gzip` and the server sends a gzip compressed response back?

Reply:
@jon-wei 
+1 This is supposed to be done by existing method `testGzipCompression()`, I added some content checking asserts to that method.

Also changed the relevant test method name to avoid confusing.